libnmstate/nm/active_connection.py

```diff
@@ -205,3 +197,23 @@ def _is_device_master_type(nmdev):
         )
         return is_master_type
     return False
+
+
+def _ip_config_can_be_treat_as_activated(nm_dev, nm_ac):
+    """
+    For interface meet any condition below will be
+    treated as activated when reach IP_CONFIG state:
+        * Is master device.
+        * DHCPv4 enabled with IPv6 disabled.
+        * DHCPv6/Autoconf enabled with IPv4 disabled.
+        * DHCPv4 enabled with DHCPv6/Autoconf enabled.
+    """
+    if _is_device_master_type(nm_dev):
+        return True
+    ac_state_flags = nm_ac.get_state_flags()
+    nm_state_flags = NM.ActivationStateFlags
+    if ipv4.is_static(nm_ac) and not ac_state_flags & nm_state_flags.IP4_READY:
+        return False
+    if ipv6.is_static(nm_ac) and not ac_state_flags & nm_state_flags.IP6_READY:
+        return False
+    return ipv4.is_dynamic(nm_ac) or ipv6.is_dynamic(nm_ac)
```

Comment:
We can return `True` directly, right? If ipv4 and ipv6 are disable this function is returning `False` which is not okay.